### DISCUSSION TITLE
How do I blacklist download mirror selection by patterns or domain?

### DISCUSSION THREAD

**Discussion by Tony-Goat** [microsoft/vcpkg#30164] [Q&A]
I'm trying to install the libmysql package, however, the download is getting stuck. After doing some digging, I figured out that it's getting stuck on a dependency, libtool, and specifically getting stuck when it's trying to download from a Chinese mirror, because my organization blocks traffic to and from servers in China by GeoIP due to governmental regulations.

I just want to blacklist that mirror so that I can move on with installing the package and continue with my work.

**Comment by StarGate-One** [comment]
Only way I know of is to edit `<vcpkg-root>\scripts\cmake\vcpkg_acquire_msys.cmake`, either remove or comment out the [URLs listed in lines 4 through 10](https://github.com/microsoft/vcpkg/blob/cfdeb75bb67fee475b517a2f45d8f93f2b46bdb4/scripts/cmake/vcpkg_acquire_msys.cmake#L4-#l10). You may be able to replace the ones you do not want with one or more from the [github MSYS mirrors list found here](https://github.com/msys2/MSYS2-packages/blob/master/pacman-mirrors/mirrorlist.msys) or [MSYS mirrors list here](https://www.msys2.org/dev/mirrors/).
Maybe you could open a PR for an enhancement to read the mirrors from a separate editable file instead of being hard-coded in the scripts - that way one could put in the URLs closest to them and follow company/government guidelines/restrictions.

**Comment by autoantwort** [comment]
I wounder why vcpkg is getting stuck in the first place. It should timeout and try the next mirror.

**Reply by Baardi** [reply]
I'll take that one step further and ask why vcpkg are using chinese and russian mirrors to begin with.

**Reply by dg0yt** [reply]
Why shouldn't Chinese and Russian users be able to benefit from Chinese and Russian mirrors?

**Reply by Baardi** [reply]
My main issue is that they end up getting used by people outside of China/Russia, and that these are some very authotitarian sanctioned countries, that regularly performs cyber attacks towards western countries.

I guess signatures helps for file integrity, but I can't help but feel a bit uncomfortable about using these mirrors.
